### PR TITLE
DEPR: Rolling.win_type returning freq & is_datetimelike

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

xref: https://github.com/pandas-dev/pandas/pull/38641#issuecomment-754118989
xref: https://github.com/pandas-dev/pandas/pull/38664/files#r551419130

### CODE REVIEW

File: pandas/tests/window/test_win_type.py [hunk 1]
```diff
@@ -137,6 +137,12 @@ def test_consistent_win_type_freq(arg):
         s.rolling(arg, win_type="freq")
 
 
+def test_win_type_freq_return_deprecation():
+    freq_roll = Series(range(2), index=date_range("2020", periods=2)).rolling("2s")
+    with tm.assert_produces_warning(FutureWarning):
+        freq_roll.win_type
```
Comment (jorisvandenbossche) [inline pandas/tests/window/test_win_type.py:143]:
```suggestion
        assert freq_roll.win_type == "freq"
```

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for looking into this!

Question: is there a reason that it can't keep returning "freq" for the `win_type` ? (You don't seem to have to change anything else by making `self.win_type` return "freq" again.) 
Or is it just that we don't need it internally?

File: pandas/tests/window/test_win_type.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/window/test_win_type.py]: Thanks. Done

**Comment by mroeschke** [comment]
Yes `win_type="freq"` was an internally managed flag that we don't need anymore.

Also since we document in https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.rolling.html

> If win_type=None, all points are evenly weighted; otherwise, win_type can accept a string of any scipy.signal window function.

I think its better for consistency that `self.win_type` is only a valid scipy.signal window function

**Comment by jorisvandenbossche** [comment]
OK, that makes sense.

Another change that I noticed is the value stored in `Rolling.window`:

```
In [5]: freq_roll = Series(range(2), index=date_range("2020", periods=2)).rolling("2s")

In [6]: pd.__version__
Out[6]: '1.2.0'

In [7]: freq_roll.win_type
Out[7]: 'freq'

In [8]: freq_roll.window
Out[8]: 2000000000

In [9]: freq_roll.win_freq
Out[9]: '2s'
```

vs master:

```
In [1]: freq_roll = Series(range(2), index=date_range("2020", periods=2)).rolling("2s")

In [2]: freq_roll.win_type

In [3]: freq_roll.window
Out[3]: '2s'
```

So `Rolling.window` changed from an int (nanoseconds I suppose?) to the freq string. A change that makes sense I think (because the nanoseconds are not really useful, you need it as a Timedelta / actual window or so anyway). 

But, that also makes it hard to follow the suggestion of "Check the type of self.window instead." in the deprecation warning (since this changed as well, or at least that means you need an actual pandas version check to do something different based on the pandas version, so not insurmountable in the end ;))

**Comment by jorisvandenbossche** [comment]
The doc build is failing, we seem to be checking the `win_type` when its "freq", so might need to catch the warning there (didn't check how it is triggered):

```
>>>-------------------------------------------------------------------------
Warning in /home/runner/work/pandas/pandas/doc/source/user_guide/window.rst at block ending on line 215
Specify :okwarning: as an option in the ipython:: block to suppress this message
----------------------------------------------------------------------------
/home/runner/work/pandas/pandas/pandas/core/groupby/base.py:30: FutureWarning: win_type will no longer return 'freq' in a future version. Check the type of self.window instead.
  kwargs[attr] = getattr(self, attr)
<<<-------------------------------------------------------------------------
```

**Comment by jorisvandenbossche** [comment]
Yeah, either catching/suppressing the warning, or either explicitly checking for the "win_type" attribute and in that case check the window and not include win_type if window is a time-based string? (if that's possible) Either is fine for me.

BTW, tested this branch with dask, and the tests are passing again (with warnings). And should be quite straightforward to update it to work on pandas master without warning as well.

File: pandas/core/window/rolling.py [hunk 2]
```diff
@@ -131,6 +132,32 @@ def __init__(
             )
         self.validate()
 
+    # TODO: Remove once win_type deprecation is enforced
+    def _shallow_copy(self, obj, **kwargs):
+        warnings.simplefilter("ignore", FutureWarning)
+        super()._shallow_copy(obj, **kwargs)
```
Comment (jorisvandenbossche) [inline pandas/core/window/rolling.py:138]:
```suggestion
        with warnings.catch_warnings():
            warnings.filterwarnings("ignore", "win_type", FutureWarning)
            super()._shallow_copy(obj, **kwargs)
```
this should ensure the warning filter is only changed inside that context
this should ensure

**Comment by jreback** [comment]
thanks @mroeschke

**Comment by jorisvandenbossche** [comment]
Thanks @mroeschke !

File: doc/source/whatsnew/v1.3.0.rst [hunk 3]
```diff
@@ -161,6 +161,8 @@ Deprecations
 - Deprecated :meth:`MultiIndex.is_lexsorted` and :meth:`MultiIndex.lexsort_depth` as a public methods, users should use :meth:`MultiIndex.is_monotonic_increasing` instead (:issue:`32259`)
 - Deprecated keyword ``try_cast`` in :meth:`Series.where`, :meth:`Series.mask`, :meth:`DataFrame.where`, :meth:`DataFrame.mask`; cast results manually if desired (:issue:`38836`)
 - Deprecated comparison of :class:`Timestamp` object with ``datetime.date`` objects.  Instead of e.g. ``ts <= mydate`` use ``ts <= pd.Timestamp(mydate)`` or ``ts.date() <= mydate`` (:issue:`36131`)
+- Deprecated :attr:`Rolling.win_type` returning ``"freq"`` (:issue:`38963`)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v1.3.0.rst:164]: small future note: since `pd.Rolling` does not exist, such a `` :attr:`Rolling.win_type`  `` does not work

Comment (mroeschke) [reply doc/source/whatsnew/v1.3.0.rst:164]: Ah okay thanks for the note.
